packages/slice-machine/lib/builders/SliceBuilder/index.tsx

```diff
@@ -109,18 +102,25 @@ const SliceBuilder: React.FC<SliceBuilderProps> = ({
 
   if (!variation || !sliceView) return null;
 
-  const onTakingCustomScreenshot = () => {
-    // eslint-disable-next-line @typescript-eslint/no-misused-promises
-    checkSimulatorSetup(true, async () => {
-      await generateScreenShot(
-        variation.id,
-        extendedComponent.component.from,
-        extendedComponent.component.model.name,
-        setData,
-        (screenshots) =>
-          generateSliceScreenshot(screenshots, extendedComponent.component)
-      );
-    });
+  const onTakingSliceScreenshot = () => {
+    generateSliceScreenshot(variation.id, extendedComponent.component, setData);
+  };
+
+  const onTakingSliceCustomScreenshot = (file: Blob) => {
+    generateSliceCustomScreenshot(
+      variation.id,
+      extendedComponent.component,
+      setData,
+      file
+    );
+  };
+
+  const onPushSlice = () => {
+    pushSlice(extendedComponent, onPush);
```

Comment:
can you just write the content of `onPush` (`setData(data)...`) here rather than having a separate function for it?

For redability otherwise we end up with a bunch of `onPush` `onPushSlice` and it's hard to understand what does what